partner_kind/models/res_partner.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2023 Akretion (https://www.akretion.com).
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ResPartner(models.Model):
+
+    _inherit = ["res.partner"]
+
+    kind = fields.Selection(
+        selection="_get_partner_kind",
+        string="Type",
+        help="Type of partner",
+        readonly=False,
+        compute="_compute_kind",
+        default="person",
+        compute_sudo=True,
+        required=True,
+        store=True,
+    )
+
+    def _get_partner_kind(self):
+        return [
+            ("user", "User"),
+            ("address", "Address"),
+            ("person", "Individual"),
+            ("company", "Company"),
+        ]
+
+    @api.depends("user_ids", "parent_id", "company_type")
+    def _compute_kind(self):
+
+        for rec in self.with_context(active_test=False):
+            if rec.user_ids:
```

Comment:
With context active_test=False, the kind of an active partner will stay as "user" even if the related res.user was archived. 
I recommend the following:
```suggestion
    def _compute_kind(self):
        for rec in self:
          if any(rec.user_ids.mapped("active")):
```

also add "user_ids.active" in @api.depends

Reply:
ok

Reply:
we need both user_ids and user_ids.active